pinot-core/src/test/java/com/linkedin/pinot/core/startree/TestStarTreeIntegrationTest.java

```diff
@@ -93,41 +99,70 @@ void tearDown()
    */
   @Test
   public void testHardCodedQueries() throws Exception {
-    String[] queries = new String[] {
+    String[] queries = new String[]{
         "select sum(m1) from T",
-        "select sum(m1) from T where d1 = 'd1-v2' group by d2"
+        "select sum(m1) from T where d1 = 'd1-v1'",
+        "select sum(m1) from T where d1 <> 'd1-v1'",
+        "select sum(m1) from T where d1 between 'd1-v1' and 'd1-v3'",
+        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2')",
+        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2') and d2 not in ('d2-v1')",
+        "select sum(m1) from T group by d1",
+        "select sum(m1) from T group by d1, d2",
+        "select sum(m1) from T where d1 = 'd1-v2' group by d1",
+        "select sum(m1) from T where d1 between 'd1-v1' and 'd1-v3' group by d2",
+        "select sum(m1) from T where d1 = 'd1-v2' group by d2, d3",
+        "select sum(m1) from T where d1 <> 'd1-v1' group by d2",
+        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2') group by d2",
+        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2') and d2 not in ('d2-v1') group by d3",
+        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2') and d2 not in ('d2-v1') group by d3, d4"
```

Comment:
As mentioned in the comment, the test runs aggregation on all metric columns, not just the one in the query, to be more exhaustive.